README.md

```diff
@@ -1,20 +1,408 @@
+profiler, err := profiles.New("myapp", profiles.WithFileStore(systemDir))
+
+// Hierarchy: 
+// 1. /Library/Managed Preferences/com.company.myapp.plist (if exists, read-only)
+// 2. /Library/Application Support/com.company/com.company.myapp/ (user-writable)
+```
+
+#### Option 2: Explicit MDM Identifier
+
+For applications with regular namespace but need MDM:
+
+```go
+// Regular namespace with explicit MDM identifier
+plat, err := platform.NewPlatform("company", "myapp", runtime.GOOS)
+systemDir, opts := plat.SystemAppDataDirectoryWithMDM("com.company.myapp")
+profiler, err := profiles.New("myapp", profiles.WithFileStore(systemDir, opts...))
+
+// Hierarchy:
+// 1. /Library/Managed Preferences/com.company.myapp.plist (if exists, read-only) 
+// 2. /Library/Application Support/company/myapp/ (user-writable)
+```
+
+#### Option 3: User Directory (No MDM)
+
+For user-only configurations:
+
+```go
+// User directory never checks MDM
+plat, err := platform.NewPlatform("com.company", "myapp", runtime.GOOS)
+userDir := plat.UserAppDataDirectory() // No MDM checking
+profiler, err := profiles.New("myapp", profiles.WithFileStore(userDir))
+
+// Uses only: ~/Library/Application Support/com.company/myapp/
+```
+
+### MDM Behavior & Error Handling
+
+| Scenario | MDM File Exists | Read Behavior | Write/Delete Behavior |
+|----------|----------------|---------------|----------------------|
+| **Key in MDM** | ✅ Yes | Returns MDM value | ❌ OS error → wrapped as "managed remotely (MDM)" |
+| **Key not in MDM** | ✅ Yes | Falls back to file storage | ✅ Writes to file storage |
+| **No MDM file** | ❌ No | Uses file storage only | ✅ Writes to file storage |
+| **Non-macOS** | ❌ N/A | Uses file storage only | ✅ Writes to file storage (or "read-only" errors) |
+
+**OS-Native Error Handling:**
+- Let the OS enforce permissions naturally (no explicit blocking)
+- Wrap OS write errors with meaningful context for downstream apps
+- Same pattern works for Windows ProgramFiles, Linux system directories
+- Apps can catch specific error types and show user-friendly messages
+
+**Example Error Types:**
+```go
+// Downstream apps can handle these specifically
+if errors.Is(err, profiles.ErrManagedByMDM) {
+    showMessage("Settings are managed by your IT department")
+} else if errors.Is(err, profiles.ErrReadOnlyLocation) {
+    showMessage("Cannot save to this location - insufficient permissions")
+}
+```
+
+### Enterprise Example
+
+```go
+// Enterprise application setup with automatic MDM
+plat, err := platform.NewPlatform("com.acme", "com.acme.secure-app", runtime.GOOS)
+systemDir := plat.SystemAppDataDirectory() // Auto-enables MDM
+profiler, err := profiles.New("secure-app", profiles.WithFileStore(systemDir))
+
+// IT Admin deploys this plist to all devices:
+// /Library/Managed Preferences/com.acme.secure-app.plist
+// {
+//   "server-url": "https://prod.acme.com",
+//   "compliance-mode": true,
+//   "user-override-allowed": false
+// }
```

Comment:
This is intended to capture any admin-only profile info that users of a device are expected to be unable to manipulate, correct?

Reply:
That's correct. I would classify that as a read-only config vs read-write.